src/code/z_collision_check.c

```diff
@@ -1,392 +1,3345 @@
+            return;
+        }
+        Audio_PlaySoundGeneral(NA_SE_IT_SHIELD_BOUND, &collider->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        return;
+    }
+    if (flags == 0x10) {
+        func_80029CA4(globalCtx, 1, arg3);
+        if (collider->actor == NULL) {
+            Audio_PlaySoundGeneral(NA_SE_IT_REFLECTION_WOOD, &D_801333D4, 4, &D_801333E0, &D_801333E0, &D_801333E8);
+            return;
+        }
+        Audio_PlaySoundGeneral(NA_SE_IT_REFLECTION_WOOD, &collider->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                               &D_801333E8);
+        return;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005E800.s")
+s32 func_8005E4F8(Collider* left, ColliderBody* rightBody) {
+    if (left->actor != NULL) {
+        if (ACTORTYPE_PLAYER == left->actor->type) {
+            if (rightBody->flags == 0) {
+                Audio_PlaySoundGeneral(NA_SE_IT_SWORD_STRIKE, &left->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            } else if (rightBody->flags == 1) {
+                Audio_PlaySoundGeneral(NA_SE_IT_SWORD_STRIKE_HARD, &left->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            } else if (2 == rightBody->flags) {
+                Audio_PlaySoundGeneral(NA_SE_PL_WALK_GROUND, &left->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            } else if (rightBody->flags == 3) {
+                Audio_PlaySoundGeneral(NA_SE_PL_WALK_GROUND, &left->actor->unk_E4, 4, &D_801333E0, &D_801333E0,
+                                       &D_801333E8);
+            }
+        }
+    }
+    return 1;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005E81C.s")
+typedef struct {
+    u8 unk00;
+    u8 unk01;
+} D_8011DF40_s;
+
+void func_8005E604(GlobalContext* globalCtx, Collider* left, ColliderBody* leftBody, Collider* right,
+                   ColliderBody* rightBody, Vec3f* arg5) {
+
+    static void (*D_8011DF28[])(GlobalContext*, Collider*, Vec3f*) = {
+        func_8005DF9C, func_8005DFAC, func_8005E10C, func_8005E26C, func_8005E2A4, func_8005E2C8,
+    };
+
+    static D_8011DF40_s D_8011DF40[0x0E] = {
+        { 1, 0 }, { 0, 1 }, { 2, 1 }, { 0, 0 }, { 3, 5 }, { 0, 2 }, { 2, 0 },
+        { 4, 0 }, { 1, 2 }, { 0, 3 }, { 0, 5 }, { 0, 3 }, { 0, 3 }, { 0, 4 },
+    };
+
+    if (rightBody->bumperFlags & 0x40) {
+        return;
+    }
+    if (!(leftBody->toucherFlags & 0x20) && leftBody->toucherFlags & 0x40) {
+        return;
+    }
+    if (right->actor != NULL) {
+        (*D_8011DF28[D_8011DF40[right->unk_14].unk00])(globalCtx, right, arg5);
+    }
+    if (right->actor != NULL) {
+        if (D_8011DF40[right->unk_14].unk01 == 3) {
+            func_8005E2EC(globalCtx, leftBody, right, arg5);
+            return;
```

Comment:
Most returns in this function can be omitted and probably should be, tho some will require else cases, like the last one for Audio_PlaySoundGeneral calls.